### PR TITLE
feat: auto rotate ads

### PR DESCRIPTION
**Pull request by omBratteng**
## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

AS-733

### Preview domain
https://as-733-rotate-ads.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/hooks/feed/useAutorotatingAds.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import { featureAutorotateAds } from '../../lib/featureManagement';
+import type { Ad } from '../../graphql/posts';
+import { apiUrl } from '../../lib/config';
+import { generateAdLogEventKey } from './useLogImpression';
+import { disabledRefetch } from '../../lib/func';
+import { RequestKey } from '../../lib/query';
+
+export type InViewRef = InViewHookResponse['ref'];
+
+export const useAutorotatingAds = (
```
Comment (idoshamun) [inline packages/shared/src/hooks/feed/useAutorotatingAds.ts:24]: I got triggered 🙈 (also the file name and everywhere else)

```suggestion
export const useAutoRotatingAds = (
```

Comment (omBratteng) [reply packages/shared/src/hooks/feed/useAutorotatingAds.ts:24]: 😂

Comment (omBratteng) [reply packages/shared/src/hooks/feed/useAutorotatingAds.ts]: Fixed

File: packages/shared/src/hooks/feed/useAutorotatingAds.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    },
+    [inViewRef, ref],
+  );
+
+  const queryKey = useMemo(
+    () => [RequestKey.Ads, ...feedQueryKey],
+    [feedQueryKey],
+  );
+
+  const fetchNewAd = useCallback(async (): Promise<Ad> => {
```
Comment (idoshamun) [inline packages/shared/src/hooks/feed/useAutorotatingAds.ts:54]: Why don't we reuse or encapsulate ad fetching? I don't want it to happen in multiple places if possible

File: packages/shared/src/lib/featureManagement.ts [hunk 3]
```diff
@@ -48,4 +48,6 @@ const feature = {
   }),
 };
 
+export const featureAutorotateAds = new Feature('autorotate_ads', 1);
```
Comment (idoshamun) [inline packages/shared/src/lib/featureManagement.ts:51]: Default should be null/undefined right? It should be turned off by default

Comment (omBratteng) [reply packages/shared/src/lib/featureManagement.ts:51]:
```suggestion
export const featureAutorotateAds = new Feature('autorotate_ads', 0);
```

Comment (omBratteng) [reply packages/shared/src/lib/featureManagement.ts]: Default is 0 supposed to be 0

**Review by idoshamun — COMMENTED** [review]
Have you tested carefully the functionality and events that are sent?

File: packages/shared/src/lib/featureManagement.ts (hunk 3 above)
Comment (omBratteng) [reply packages/shared/src/lib/featureManagement.ts]: You control the duration of how long to render each ad (in ms)

File: packages/shared/src/hooks/feed/useAutorotatingAds.ts (hunk 1 above)
Comment (idoshamun) [reply packages/shared/src/hooks/feed/useAutorotatingAds.ts]: File name is still lower case 🙈

File: packages/shared/src/hooks/feed/useAutorotatingAds.ts (hunk 2 above)
Comment (omBratteng) [reply packages/shared/src/hooks/feed/useAutorotatingAds.ts:54]: Refactored it to we have one function for calling ads

**Review by idoshamun — APPROVED** [review]

File: packages/shared/src/hooks/feed/useAutorotatingAds.ts (hunk 1 above)
Comment (omBratteng) [reply packages/shared/src/hooks/feed/useAutorotatingAds.ts]: Now it should be fixed...

File: packages/shared/src/hooks/feed/useAutorotatingAds.ts (hunk 2 above)
Comment (omBratteng) [reply packages/shared/src/hooks/feed/useAutorotatingAds.ts]: Also refactored the comment ad fetch, so it's in same utils file

**Pull request merged**